java/src/main/native/src/CudaJni.cpp

```diff
@@ -195,6 +195,32 @@ JNIEXPORT jint JNICALL Java_ai_rapids_cudf_Cuda_getNativeComputeMode(JNIEnv *env
   CATCH_STD(env, -2);
 }
 
+namespace {
+
+jint cuda_device_get_attribute_impl(JNIEnv *env, cudaDeviceAttr const &attribute_label) {
+  try {
+    cudf::jni::auto_set_device(env);
+    int device;
+    JNI_CUDA_TRY(env, -2, ::cudaGetDevice(&device));
```

Comment:
Humm, I suspect that such cuda call has some (significant?) overhead. Should we combine the `*Major` and `*Minor` methods into one function that returns both?

Reply:
HMM. Not if we just want the major version, for instance, to identify Pascal devices. 
I did think about that though. I didn't want to have to deal with piercing the JNI boundary with a `pair`, or heap allocations. 

This is in service of the linked `spark-rapids` bug on Pascal, FWIW.

Reply:
I believe these CUDA calls are very cheap.

Reply:
I might have misread Nghia's comment. It might have been regarding crossing the JNI boundary twice, for major and minor versions. 
That cost is unavoidable if we need to free the heap allocation for an `int[2]`. 

If the comment was regarding the CUDA call itself, I'm with Jason: I don't think it's all that expensive.